nx/lib/nx.ex

```diff
@@ -12448,13 +12448,27 @@ defmodule Nx do
     feature_group_count = opts[:feature_group_size]
     batch_group_count = opts[:batch_group_size]
 
-    %{shape: input_shape, names: input_names} = tensor = to_tensor(tensor)
-    %{shape: kernel_shape, names: kernel_names} = kernel = to_tensor(kernel)
+    [tensor, kernel] = broadcast_vectors([tensor, kernel], align_ranks: false)
+    Nx.Shape.validate_conv!(tensor.shape, kernel.shape)
 
-    Nx.Shared.raise_vectorized_not_implemented_yet(tensor, __ENV__.function)
-    Nx.Shared.raise_vectorized_not_implemented_yet(kernel, __ENV__.function)
+    vectorized_axes = tensor.vectorized_axes
+    offset = length(vectorized_axes)
+
+    %{shape: t_shape} = tensor = devectorize(tensor)
+
+    {batch_axes, other_axes} = t_shape |> Tuple.to_list() |> Enum.split(offset + 1)
+    {_, names} = Enum.split(tensor.names, offset)
+
+    tensor = reshape(tensor, List.to_tuple([Enum.product(batch_axes) | other_axes]), names: names)
+
+    %{shape: k_shape} = kernel = devectorize(kernel)
+    {_, names} = Enum.split(kernel.names, offset)
+    {batch_axes, other_axes} = k_shape |> Tuple.to_list() |> Enum.split(offset + 1)
 
-    Nx.Shape.validate_conv!(input_shape, kernel_shape)
+    kernel = reshape(kernel, List.to_tuple([Enum.product(batch_axes) | other_axes]), names: names)
```

Comment:
I think we can encapsulate this in a function, so we share between input and kernel. Also, I understand that the reshape `shape` will match the tensor shape and be a no-op for the common case, but I think it is clearer if we perform this code path only if the tensor is vectorized. :)

Reply:
By this, I mean this code:

```
    %{shape: k_shape} = kernel = devectorize(kernel)
    {_, names} = Enum.split(kernel.names, offset)
    {batch_axes, other_axes} = k_shape |> Tuple.to_list() |> Enum.split(offset + 1)

    Nx.Shape.validate_conv!(input_shape, kernel_shape)
    kernel = reshape(kernel, List.to_tuple([Enum.product(batch_axes) | other_axes]), names: names)
```

It seems to be shared between both input and kernel.